src/Keboola/DbExtractor/Extractor/Firebird.php

```diff
@@ -50,7 +51,66 @@ public function testConnection(): void
 
     public function getTables(?array $tables = null): array
     {
-        throw new UserException('This component does not yet support the getTables method');
+        $resultTables = $this->runRetriableQuery(
+            "SELECT TRIM(RDB\$RELATION_NAME) AS NAME,
+				CASE RDB\$VIEW_BLR WHEN NULL THEN 'TRUE' ELSE 'FALSE' END AS VIEW_CASE
+			FROM RDB\$RELATIONS
+			WHERE RDB\$SYSTEM_FLAG is null or RDB\$SYSTEM_FLAG = 0;"
+        );
+        $tables = [];
+        foreach ($resultTables as $table) {
+            $tables[] = [
+                'name' => $table['NAME'],
+                'view' => $table['VIEW_CASE'] === 'TRUE',
+                'columns' => $this->getTableColumns($table['NAME']),
+            ];
+        }
+        return $tables;
+    }
+
+    private function getTableColumns(string $table): array
+    {
+        $table = strtoupper($table);
+        $resultColumns = $this->runRetriableQuery(
+            "SELECT TRIM(r.RDB\$FIELD_NAME) AS FIELD_NAME,
+				CASE f.RDB\$FIELD_TYPE
+					WHEN 261 THEN 'BLOB'
+					WHEN 14 THEN 'CHAR'
+					WHEN 40 THEN 'CSTRING'
+					WHEN 11 THEN 'D_FLOAT'
+					WHEN 27 THEN 'DOUBLE'
+					WHEN 10 THEN 'FLOAT'
+					WHEN 16 THEN 'INT64'
+					WHEN 8 THEN 'INTEGER'
+					WHEN 9 THEN 'QUAD'
+					WHEN 7 THEN 'SMALLINT'
+					WHEN 12 THEN 'DATE'
+					WHEN 13 THEN 'TIME'
+					WHEN 35 THEN 'TIMESTAMP'
+					WHEN 37 THEN 'VARCHAR'
+					ELSE 'UNKNOWN'
+				END AS FIELD_TYPE,
+				f.RDB\$FIELD_LENGTH AS FIELD_LENGTH,
+				CASE r.RDB\$NULL_FLAG
+					WHEN 1 THEN 'FALSE' ELSE 'TRUE'
+				END AS NULLABLE
+			FROM RDB\$RELATION_FIELDS r
+				LEFT JOIN RDB\$FIELDS f ON r.RDB\$FIELD_SOURCE = f.RDB\$FIELD_NAME
+			WHERE r.RDB\$RELATION_NAME = '$table'
+			ORDER BY r.RDB\$FIELD_POSITION;"
+        );
+        $columns = [];
+        foreach ($resultColumns as $column) {
+            $key = $column['FIELD_NAME'];
+            $baseType = new GenericStorage(trim($column['FIELD_TYPE']));
```

Comment:
I think it's great, but I'm afraid that there may be a couple of missing types in the `GenericStorage` class that could result in some false matching.  For example I don't think `GenericStorage` contains `D_FLOAT` or `QUAD` . (I actually don't know what quad is:).

We can either change for example `D_FLOAT` to `FLOAT` here, or add them to the `GenericStorage` class.  To necham na tobe.

Reply:
@pivnicek ok so i add PR in php-datatypes - https://github.com/keboola/php-datatypes/pull/25

`quad` is `Quadruple precision floating point format` => float :)